### PR TITLE
Fixed: Style and lint violations in src/bitmessageqt/__init__.py

### PR DESCRIPTION
**Pull request by coffeedogs**
Fixed: Style and lint violations in src/bitmessageqt/__init__.py

### CODE REVIEW

File: src/bitmessageqt/__init__.py [hunk 1]
```diff
@@ -1,63 +1,78 @@
...
-import debug
-import random
+
 from sqlite3 import register_adapter
-import string
-from datetime import datetime, timedelta
-from helper_ackPayload import genAckPayload
-from helper_sql import sqlQuery, sqlExecute, sqlExecuteChunked, sqlStoredProcedure
+
+import debug  # pylint: disable=ungrouped-imports
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: Please place imports at the starting and in a order

File: src/bitmessageqt/__init__.py [hunk 2]
```diff
@@ -961,9 +1024,12 @@ def updateUnreadStatus(self, widget, row, msgid, unread=True):
...
                     related.item(rrow, col).setUnread(not status)
 
-    def propagateUnreadCount(self, address = None, folder = "inbox", widget = None, type = 1):
+    def propagateUnreadCount(self, address=None, folder="inbox", widget=None, type_arg=1):
+        """TBC"""
+
         widgets = [self.ui.treeWidgetYourIdentities, self.ui.treeWidgetSubscriptions, self.ui.treeWidgetChans]
-        queryReturn = sqlQuery("SELECT toaddress, folder, COUNT(msgid) AS cnt FROM inbox WHERE read = 0 GROUP BY toaddress, folder")
+        queryReturn = sqlQuery(
+            "SELECT toaddress, folder, COUNT(msgid) AS cnt FROM inbox WHERE read = 0 GROUP BY toaddress, folder")
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: use pep8

File: src/bitmessageqt/__init__.py [hunk 3]
```diff
@@ -1007,24 +1076,29 @@ def propagateUnreadCount(self, address = None, folder = "inbox", widget = None,
                     if addressItem.type == AccountMixin.ALL and folderName in totalUnread:
                         newCount = totalUnread[folderName]
                     elif addressItem.type == AccountMixin.SUBSCRIPTION:
-                        if addressItem.address in broadcastsUnread and folderName in broadcastsUnread[addressItem.address]:
+                        if addressItem.address in broadcastsUnread and folderName in broadcastsUnread[
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: use pep8

File: src/bitmessageqt/__init__.py [hunk 4]
```diff
@@ -2411,32 +2764,57 @@ def click_actionSettings(self):
...
+            if int(BMConfigParser().get('bitmessagesettings', 'port')) != int(
+                    self.settingsDialogInstance.ui.lineEditTCPPort.text()):
                 if not BMConfigParser().safeGetBoolean('bitmessagesettings', 'dontconnect'):
                     QtGui.QMessageBox.about(self, _translate("MainWindow", "Restart"), _translate(
                         "MainWindow", "You must restart Bitmessage for the port number change to take effect."))
                 BMConfigParser().set('bitmessagesettings', 'port', str(
                     self.settingsDialogInstance.ui.lineEditTCPPort.text()))
-            if self.settingsDialogInstance.ui.checkBoxUPnP.isChecked() != BMConfigParser().safeGetBoolean('bitmessagesettings', 'upnp'):
-                BMConfigParser().set('bitmessagesettings', 'upnp', str(self.settingsDialogInstance.ui.checkBoxUPnP.isChecked()))
+            if self.settingsDialogInstance.ui.checkBoxUPnP.isChecked() != BMConfigParser().safeGetBoolean(
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: Pep8 validation

File: src/bitmessageqt/__init__.py [hunk 5]
```diff
@@ -2411,32 +2764,57 @@ def click_actionSettings(self):
...
+                            'bitmessagesettings',
+                            'socksproxytype') == 'none',
+                        self.settingsDialogInstance.ui.comboBoxProxyType.currentText()[0:5] == 'SOCKS',
+                    ]
+            ):
                 if shared.statusIconColor != 'red':
-                    QtGui.QMessageBox.about(self, _translate("MainWindow", "Restart"), _translate(
-                        "MainWindow", "Bitmessage will use your proxy from now on but you may want to manually restart Bitmessage now to close existing connections (if any)."))
-            if BMConfigParser().get('bitmessagesettings', 'socksproxytype')[0:5] == 'SOCKS' and self.settingsDialogInstance.ui.comboBoxProxyType.currentText()[0:5] != 'SOCKS':
+                    QtGui.QMessageBox.about(
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: pep8 validation

File: src/bitmessageqt/__init__.py [hunk 6]
```diff
@@ -2480,46 +2858,85 @@ def click_actionSettings(self):
...
 
             acceptableDifficultyChanged = False
-            
-            if float(self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()) >= 1 or float(self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()) == 0:
-                if BMConfigParser().get('bitmessagesettings','maxacceptablenoncetrialsperbyte') != str(int(float(
-                    self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()) * defaults.networkDefaultProofOfWorkNonceTrialsPerByte)):
+
+            if any(
+                    [
+                        float(self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()) >= 1,
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: pep8 validation

File: src/bitmessageqt/__init__.py [hunk 7]
```diff
@@ -2480,46 +2858,85 @@ def click_actionSettings(self):
...
-                        self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()) * defaults.networkDefaultProofOfWorkNonceTrialsPerByte)))
-            if float(self.settingsDialogInstance.ui.lineEditMaxAcceptableSmallMessageDifficulty.text()) >= 1 or float(self.settingsDialogInstance.ui.lineEditMaxAcceptableSmallMessageDifficulty.text()) == 0:
-                if BMConfigParser().get('bitmessagesettings','maxacceptablepayloadlengthextrabytes') != str(int(float(
-                    self.settingsDialogInstance.ui.lineEditMaxAcceptableSmallMessageDifficulty.text()) * defaults.networkDefaultPayloadLengthExtraBytes)):
+                        self.settingsDialogInstance.ui.lineEditMaxAcceptableTotalDifficulty.text()
+                    ) * defaults.networkDefaultProofOfWorkNonceTrialsPerByte)))
+
+            if any(
+                    [
+                        float(self.settingsDialogInstance.ui.lineEditMaxAcceptableSmallMessageDifficulty.text()) >= 1,
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: pep8 validation

**Comment by coffeedogs** [comment]
@omkar1117 regarding the variable name violations, we are specifically ignoring them for now. At some point we'll do a big-bang fix but we can't do that piecemeal like the other style issues since they permeate the entire project.

File: src/bitmessageqt/__init__.py (hunk 1 above)
Comment (coffeedogs) [reply src/bitmessageqt/__init__.py]: Sadly impossible without changing functionality. We need our `logger` during the import of a third party module.

In the next commit I have grouped the debugs, placed them after the built-in imports and now there is only one marked as violating, which is as much as can be done.

File: src/bitmessageqt/__init__.py (hunk 6 above)
Comment (coffeedogs) [reply src/bitmessageqt/__init__.py]: This is actually correct. All tools pass this and actually require four extra spaces here if you take them out. It's to do with the visual indentation compared to the contents of the conditional. Once the fab work is in you can run:
```
coffeedogs@tinternet:~/PyBitmessageSplit2$ fab -H localhost code_quality:fix=0,details=1,filename=src/bitmessageqt/__init__.py
[localhost] Executing task 'code_quality'          
                                     
total pycodestyle flake8 pylint path_to_file                               
                                                     
0 0 0 0 /home/coffeedogs/PyBitmessageSplit2/src/bitmessageqt/__init__.py               
pycodestyle:    
                                  
flake8:                                                    
                                                                                                    
pylint:                                                
                                                      
Disconnecting from localhost... done.
```

Comment (g1itch) [reply src/bitmessageqt/__init__.py]: May it be like
```python
if any([
    float(self.settingsDialogInstance.ui...
    ...
]):
```
?

**Comment by PeterSurda** [comment]
@coffeedogs How thoroughly have you test these changes (by using the UI)?

File: src/bitmessageqt/__init__.py (hunk 6 above)
Comment (coffeedogs) [reply src/bitmessageqt/__init__.py]: That causes a pylint warning:
```C:604, 0: Wrong hanging indentation before block (add 4 spaces).```
Personally I would be happy with the style you suggest. Disabling just this case without disabling all hanging indentation warnings would mean forking pylint though.

File: src/bitmessageqt/__init__.py [hunk 8]
```diff
@@ -18,6 +16,9 @@
 import time
 from datetime import datetime, timedelta
```
Comment (omkar1117) [inline src/bitmessageqt/__init__.py]: @coffeedogs can you write import statements first and then go with from statements.

File: src/bitmessageqt/__init__.py [hunk 9]
```diff
@@ -1007,24 +1077,29 @@ def propagateUnreadCount(self, address = None, folder = "inbox", widget = None,
...
 
     def addMessageListItem(self, tableWidget, items):
+        """TBC"""
+
         sortingEnabled = tableWidget.isSortingEnabled()
         if sortingEnabled:
             tableWidget.setSortingEnabled(False)
         tableWidget.insertRow(0)
-        for i in range(len(items)):
+        for i, _ in enumerate(items):
```
Comment (PeterSurda) [inline src/bitmessageqt/__init__.py]: :+1:

**Pull request merged**